vllm/model_executor/models/mllama.py

```diff
@@ -87,40 +88,57 @@ def _get_num_image_in_last_group(prompt_token_ids: List[int]) -> int:
 
 
 def input_processor_for_mllama(ctx: InputContext,
-                               inputs: Union[DecoderOnlyInputs,
-                                             EncoderDecoderInputs]):
-    # move encoder_prompt to prompt
-    if inputs.get("prompt") is None:
-        inputs["prompt"] = inputs["encoder_prompt"]
-        inputs["prompt_token_ids"] = inputs["encoder_prompt_token_ids"]
-
-    # process multi-modal data
-    multi_modal_data = inputs.get("encoder_multi_modal_data")
-
-    if multi_modal_data is None or "image" not in multi_modal_data \
-        or multi_modal_data["image"] is None:
+                               inputs: EncoderDecoderInputs):
+    # Example input to processor:
+    # {
+    #     'encoder': {
+    #         'type': 'token',
+    #         'prompt_token_ids': [128000, 128256, 128000, 3923, 374, 279, 2262, 315, 420, 2217, 30],  # noqa: E501
+    #         'prompt': '<|image|><|begin_of_text|>What is the content of this image?',  # noqa: E501
+    #         'multi_modal_data': {'image': <PIL.Image.Image image mode=RGB size=1770x1180 at 0x7FDE2C624880>},  # noqa: E501
+    #     },
+    #     'decoder': {
+    #         'type': 'token',
+    #         'prompt_token_ids': [128000],
+    #     },
+    # }
+
+    # move encoder prompt to decoder
+    inputs["decoder"] = TokenInputs(**inputs["encoder"])
+
+    dec_inputs = inputs["decoder"]
```

Comment:
Does inputs really need to be updated here?

Couldn't you just do
```python
dec_inputs = inputs["encoder"]
```
?

Reply:
That's correct, thanks for simplifying!